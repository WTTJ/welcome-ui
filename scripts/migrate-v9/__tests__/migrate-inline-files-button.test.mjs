import { readFileSync } from 'fs'
import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'

import { describe, expect, it } from 'vitest'

import { copyDirSync, deleteDirRecursive } from '../helpers/file-utils.mjs'
import { migrateAll } from '../index.mjs'

const __dirname = dirname(fileURLToPath(import.meta.url))

describe('Inline Migration - Button Components', () => {
  it('migrates Button components with styling and form props to Button with classNames', async () => {
    const fixturesDir = resolve(__dirname, './__fixtures__')
    const tempDir = resolve(__dirname, '../temp/inline-button-test')

    // Clean up and copy fixtures
    deleteDirRecursive(tempDir)
    copyDirSync(fixturesDir, tempDir)

    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'ButtonComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('button-component.tsx')

    // Cleanup
    deleteDirRecursive(tempDir)
  })
})
