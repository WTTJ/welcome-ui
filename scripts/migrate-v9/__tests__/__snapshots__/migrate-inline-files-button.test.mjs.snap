// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Button Components > migrates Button components with styling and form props to Button with classNames > button-component.tsx 1`] = `
"/* eslint-disable no-console */
import { Button } from 'welcome-ui/Button'

const FormattedMessage = ({
  defaultMessage,
  id,
}: {
  defaultMessage: string
  id: string
}): React.ReactNode => <span id={id}>{defaultMessage}</span>

const SIGN_IN_PATH = '/signin'
const FALSE = false
const ON_CLICK = () => console.debug('submit')

export const ButtonComponent = () => {
  return (
    <form>
      <Button data-testid="my-draft-button" disabled={FALSE} onClick={ON_CLICK} size="sm">
        Published
      </Button>
      <Button as="a" className="mt-xxl w-full" dataTestId="my-button-1" to={SIGN_IN_PATH}>
        <FormattedMessage defaultMessage="Back to Sign in" id="D7hbFB" />
      </Button>
      <Button dataTestId="my-button-2" onClick={() => console.debug('submit')} type="submit">
        <FormattedMessage defaultMessage="Back to Sign in" id="D7hbFB" />
      </Button>
    </form>
  )
}
"
`;
